lib/miq_expression/invalid.rb

```diff
@@ -0,0 +1,21 @@
+class MiqExpression::Invalid < MiqExpression::Target
```

Comment:
`MiqExpression::Invalid` makes it sound like an invalid _expression_ as opposed to an invalid _target_.  This might be clearer:

```suggestion
class MiqExpression::InvalidTarget < MiqExpression::Target
```